### PR TITLE
units support single-tap move

### PR DESCRIPTION
**Pull request by sulai**
I'm working towards better user experience using a simple rule: save clicks. So I tried to implement single-tap move on units, which works out pretty well. Moving units becomes quite a bit less tedious. How it works:

"Whenever a movement can be done in 1 turn, do it immediately, without showing the move-to button."

Other changes to make this work smoothly:

- units stay selected after move (to go on moving)
- and they are de-selectable by taping them again (to abort moving)

Let me know what you think about this idea =)

### CODE REVIEW

**Comment by yairm210** [comment]
What about when clicking on enemies?

**Comment by sulai** [comment]
Attacking enemies works as before, nothing changed there. However this PR is not yet finished I think. I played a test game and I accidentally moved units every now and then. I think the movement-indicators should stay out stronger to help with that. So this one is a little experimental yet I guess.

**Comment by sulai** [comment]
This PR now depends a little on #754, where you can't click through windows anymore, avoiding accidental movement

**Comment by sulai** [comment]
This PR includes the changes of #757, as it pretty much depends on them.
Single tap movement works pretty well now. If you like, give it a try! =)

**Comment by sulai** [comment]
@yairm210 you can try all my patches (PRs) together in my integration branch: https://github.com/sulai/UnCiv/tree/integration

**Comment by yairm210** [comment]
I think this is a good idea, but the potential for harm inn the current implementation is larger that the potential benefit
One thing we could do, is take a page from Android's OS, which has a secondary action on long click.
What this would mean in our context is that when clicking a tile, a ring would start to build up, and after e.g. 500 ms that unit would move there
It's replacing 2 click with a single longer one, which I think might be easier for players, but it's difficult for me to know =\

**Comment by sulai** [comment]
The "beauty" of single tap move is that it's quick and effortless. However, it's a bit hard for beginners I think, because they might want to examine tiles more than moving quick. Maybe another entry in the options menu is a better way to go: "quick move units: yes/no", defaulting to no to give newbies a smooth start, and advanced players may opt in to more comfort. Adcanced players might also know that it's easy to de-select units by taping them again, and in that way aborting quick move.

**Comment by yairm210** [comment]
OK, sounds like a good 'extra option' for advanced players
Please add the necessary setting option and fix the conflicts, and I'll merge it

**Comment by sulai** [comment]
I added the options entry and double checked the PR to be clean. There were some left overs from my previous merge. I either should remember to re-merge dependent branches or not merge them at all :|

Anyway it looks good to me now :-) If you feel confident about this PR, feel free to merge.

**Comment by yairm210** [comment]
Apparently you can't select units in tiles anymore -_-

File: core/src/com/unciv/ui/worldscreen/TileMapHolder.kt [hunk 1]
```diff
@@ -79,10 +81,14 @@ class TileMapHolder(internal val worldScreen: WorldScreen, internal val tileMap:
         selectedTile = tileInfo
 
         val selectedUnit = worldScreen.bottomBar.unitTable.selectedUnit
+        worldScreen.bottomBar.unitTable.tileSelected(tileInfo)
```
Comment (yairm210) [inline core/src/com/unciv/ui/worldscreen/TileMapHolder.kt]: This was the problem, we now call tileSelected TWICE, once here

File: core/src/com/unciv/ui/worldscreen/TileMapHolder.kt [hunk 2]
```diff
@@ -98,20 +104,31 @@ class TileMapHolder(internal val worldScreen: WorldScreen, internal val tileMap:
         }
 
         worldScreen.bottomBar.unitTable.tileSelected(tileInfo)
```
Comment (yairm210) [inline core/src/com/unciv/ui/worldscreen/TileMapHolder.kt]: And once HERE

**Comment by yairm210** [comment]
image

Not sure I like how the units are over the city button now.

**Comment by yairm210** [comment]
image

Here, even though some of the units are "hidden", they can now easily be revealed thanks to your guidance and ideas, so I think I'll revert the city buttons to be back above the units for now =\
I don't have a better idea how to display everything always

**Comment by yairm210** [comment]
Also if you zoom out you'll be able to see the units now that I think about it

File: core/src/com/unciv/ui/worldscreen/TileMapHolder.kt (hunk 2 above)
Comment (sulai) [reply core/src/com/unciv/ui/worldscreen/TileMapHolder.kt]: I'm sorry for the inconvenince. I remember having moved that line up. Not sure why it's now duplicated. Maybe I solved a merge conflict in a bad way lately? Really not sure.

**Pull request merged**